scipy/stats/_continuous_distns.py

```diff
@@ -5335,6 +5335,100 @@ def _fitstart(self, data, args=None):
 invweibull = invweibull_gen(a=0, name='invweibull')
 
 
+class jf_skew_t_gen(rv_continuous):
+    r"""Jones and Faddy skew-t distribution [1]_.
+
+    %(before_notes)s
+
+    Notes
+    -----
+    For real parameters :math:`a>0` and :math:`b>0`, the probability density
+    function for `jf_skew_t` is:
+    The probability density function for `jf_skew_t` is:
+
+    .. math::
+
+        f(t; a, b) = C_{a,b}^{-1}
+                    \left(1+\frac{t}{\left(a+b+t^2\right)^{1/2}}\right)^{a+1/2}
+                    \left(1-\frac{t}{\left(a+b+t^2\right)^{1/2}}\right)^{b+1/2}
+
+    for real numbers :math:`a>0` and :math:`b>0`, where
+    :math:`C_{a,b} = 2^{a+b-1}B(a,b)(a+b)^{1/2}`, and :math:`B` denotes the
+    beta function (`scipy.special.beta`).
+
+    When :math:`a=b` this reduces to the Student-t distribution on :math:`2a`
+    degrees of freedom. When `a < b` or `a > b`, the distribution is negatively
+    or positively skewed respectively. In fact, :math:`f(t;b,a)=f(-t;a,b)`.
+
+    `jf_skew_t` takes :math:`a` and :math:`b` as shape parameters.
+
+    %(after_notes)s
+
+    References
+    ----------
+    .. [1] M.C. Jones and M.J. Faddy. "A skew extension of the t distribution,
+           with applications" *Journal of the Royal Statistical Society*.
+           Series B (Statistical Methodology) 65, no. 1 (2003): 159-174.
+           doi: 10.1111/1467-9868.00378
+
+    %(example)s
+
+    """
+    def _pdf(self, x, a, b):
+        c = 2 ** (a + b - 1) * sc.beta(a, b) * np.sqrt(a + b)
+        d1 = (1 + x / np.sqrt(a + b + x ** 2)) ** (a + 0.5)
+        d2 = (1 - x / np.sqrt(a + b + x ** 2)) ** (b + 0.5)
+        return d1 * d2 / c
+
+    def _rvs(self, a, b, size=None, random_state=None):
+        d1 = beta.rvs(a, b, size=size, random_state=random_state)
```

Comment:
In #17624 , I noticed that using the RNG's [beta](https://numpy.org/doc/stable/reference/random/generated/numpy.random.Generator.beta.html#numpy.random.Generator.beta) function is faster than calling scipy's `beta` distribution. Could you test it here as well?